src/model.jl

```diff
@@ -483,72 +484,160 @@ function constraints_node(m, 𝒩, 𝒯, 𝒫, ℒ, modeltype::EnergyModel)
 end
 
 """
-    constraints_emissions(m, 𝒩, 𝒯, 𝒫, ℒ, modeltype::EnergyModel)
+    constraints_emissions(m, 𝒳, 𝒫, 𝒯, modeltype::EnergyModel)
 
 Create constraints for the emissions accounting for both operational and strategic periods.
 """
-function constraints_emissions(m, 𝒩, 𝒯, 𝒫, ℒ, modeltype::EnergyModel)
-    𝒩ᵉᵐ = filter(has_emissions, 𝒩)
-    ℒᵉᵐ = filter(has_emissions, ℒ)
+function constraints_emissions(m, 𝒳, 𝒫, 𝒯, modeltype::EnergyModel)
+    # Declaration of the required subsets
     𝒫ᵉᵐ = filter(is_resource_emit, 𝒫)
     𝒯ᴵⁿᵛ = strategic_periods(𝒯)
 
+    emissions = JuMP.Containers.DenseAxisArray[]
```

Comment:
I tried to specify it as an ordinary array but I got errors in the functions.
Note that based on the functions `emissions_operational`, it will always be dense.